### ISSUE TITLE
Can't use removeAnnotations with a List of PolyLines

### ISSUE THREAD

**Issue by tobrun** [mapbox/mapbox-gl-native#2811]
Behaviour below is unwanted because:
 - PolyLine extends MultiPoint
 - MulitPoint extends Annotation.

![screen shot 2015-10-27 at 10 32 33](https://cloud.githubusercontent.com/assets/2151639/10754167/39fb79de-7c96-11e5-9210-0850ca934546.png)

**Comment by tobrun** [comment]
We need to solve this using generics and a wildcard:

```java
public void removeAnnotations(@NonNull List<? extends Annotation> annotationList) {
```

**Comment by tobrun** [comment]
Will be added in #2809

**Comment by tobrun** [comment]
closed with  f3ffb45
